nkms/config/utils.py

```diff
@@ -0,0 +1,148 @@
+import json
+import os
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.hkdf import HKDF
+from cryptography.hazmat.primitives.kdf.scrypt import Scrypt
+from eth_account import Account
+from nacl.secret import SecretBox
+from umbral.keys import UmbralPrivateKey
+
+from web3.auto import w3
+w3.eth.enable_unaudited_features()
+w3.eth.account
+
+from nkms.config.config import KMSConfig, Wallet, KMSKeyring
+
+
+def validate_passphrase(passphrase) -> str:
+    """Validate a passphrase and return it or raise"""
+
+    rules = (
+        (len(passphrase) >= 16, 'Too short'),
+    )
+
+    for rule, failure_message in rules:
+        if not rule:
+            raise KMSConfig.KMSConfigurationError(failure_message)
+    else:
+        return passphrase
+
+
+def _derive_master_key_from_passphrase(salt: bytes, passphrase: str) -> bytes:
```

Comment:
passphrase - `bytes`?  Or do we really want unicode here?

Reply:
It's user provided input and it gets encoded to bytes for derivation.